shopfloor/services/location_content_transfer.py

```diff
@@ -94,38 +88,29 @@ def _response_for_start_single(self, pickings, message=None):
             message=message,
         )
 
-    def _response_for_scan_destination(self, location, next_content, message=None):
+    def _response_for_scan_destination(
+        self, location, next_content, message=None, confirmation_required=False
+    ):
         """Transition to the 'scan_destination' state
 
         The client screen shows details of the package level or move line to move.
         """
+        data = self._data_content_line_for_location(location, next_content)
+        data["confirmation_required"] = confirmation_required
+        if confirmation_required and not message:
+            message = self.msg_store.need_confirmation()
         return self._response(
-            next_state="scan_destination",
-            data=self._data_content_line_for_location(location, next_content),
-            message=message,
-        )
-
-    def _response_for_confirm_scan_destination(
-        self, location, next_content, message=None
-    ):
-        """Transition to the 'confirm_scan_destination' state
-
-        The client screen shows details of the package level or move line to
-        move. The user has to scan the destination location a second time to
-        validate the destination.
-        """
-        return self._response(
-            next_state="confirm_scan_destination",
-            data=self._data_content_line_for_location(location, next_content),
-            message=message,
+            next_state="scan_destination", data=data, message=message,
         )
 
     def _data_content_all_for_location(self, pickings):
         sorter = self.actions_for("location_content_transfer.sorter")
         sorter.feed_pickings(pickings)
         lines = sorter.move_lines()
         package_levels = sorter.package_levels()
+        location = pickings.mapped("move_line_ids.location_id")
```

Comment:
We could have more than 1 source location here, no? Should we add an ensure_one?

Reply:
hmm but this is the source location: it should always be the same because the whole thing start by scanning it, no?

Reply:
Yes, that was my point, to make things clearer and not subject to interpretation.

Reply:
I'll add an `assert`